pyo3-macros-backend/src/params.rs

```diff
@@ -161,14 +169,20 @@ pub fn impl_arg_params(
     // create array of arguments, and then parse
     (
         quote! {
-                const DESCRIPTION: #pyo3_path::impl_::extract_argument::FunctionDescription = #pyo3_path::impl_::extract_argument::FunctionDescription {
-                    cls_name: #cls_name,
-                    func_name: stringify!(#python_name),
-                    positional_parameter_names: &[#(#positional_parameter_names),*],
-                    positional_only_parameters: #positional_only_parameters,
-                    required_positional_parameters: #required_positional_parameters,
-                    keyword_only_parameters: &[#(#keyword_only_parameters),*],
-                };
+                const PARAMETER_NAMES: &[&str] = &[#(#parameter_names),*];
+                fn argument_lookup_by_name(name: &str) -> Option<usize> {
+                    PARAMETER_NAMES.iter().position(|&n| n == name)
+                }
```

Comment:
From godbolt it looks like the compiler is pretty smart about optimizing this, because `PARAMETER_NAMES` is known at compile time it'll do tricks like check the length of `name` and then search the individual bytes of `name` to generate a matching value, rather than repeated string comparisons.